### PR TITLE
Add all code from private repository

### PR DESCRIPTION
**Pull request by arxeiss**
This is copy from SDK folder. Performed changes:
- Changed Go module name to reflect repository name
- Renamed provider from `jarvis` into `indykite`
- Added more linter rules

This is just to add all code to repo. Further refactoring and publishing to Terraform Registry will be done in future sprints.

### CODE REVIEW

**Review by maprokes — APPROVED** [review]

**Review by LadislavFolta — APPROVED** [review]

**Review by bakkelim — APPROVED** [review]

**Review by marvinkite — CHANGES_REQUESTED** [review]
Looks good, there are some clean up comments.

File: example/main.tf [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+//    redirect_uri = [
+//      "https://indykite-dev.indykite.id/handler/oauth2/callback"]
+//  }
+//
+//  depends_on = [
+//    indykite_service_authentication
+//  ]
+//}
+//
+//
```
Comment (maaland) [inline example/main.tf:183]: remove comments if not relevant

File: example/main.tf [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+      "stock@chocolate-factory.com"
+    ]
+  }
+
+  # sendgrid {
+  #   api_key = "sendgrid_access_key_id_for_wonka"
+  #   sandbox_mode = true
+  #   ip_pool_name = "oompa_pool"
+  #   host = "https://wonka.sengrid.com"
+  # }
```
Comment (maaland) [inline example/main.tf:248]: remove comments if not relevant

File: indykite/provider.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+			"indykite_application_agent":            resourceApplicationAgent(),
+			"indykite_application_agent_credential": resourceApplicationAgentCredential(),
+			"indykite_auth_flow":                    resourceAuthFlow(),
+			"indykite_authorization_policy":         resourceAuthorizationPolicy(),
+			"indykite_email_notification":           resourceEmailNotification(),
+			"indykite_ingest_mapping":               resourceIngestMapping(),
+			"indykite_oauth2_client":                resourceOAuth2Client(),
+			"indykite_oauth2_provider":              resourceOAuth2Provider(),
+			"indykite_oauth2_application":           resourceOAuth2Application(),
+			// "indykite_webauthn_provider":            resourceWebAuthnProvider(),
```
Comment (maaland) [inline indykite/provider.go:79]: remove comment if not relevant

**Review by maaland — CHANGES_REQUESTED** [review]
Some minor nitpicks, otherwise it LGTM

File: README.md [hunk 4]
```diff
@@ -1 +1,96 @@
-# terraform-provider-indykite
+<a href="https://terraform.io">
+    <img src="https://cdn.rawgit.com/hashicorp/terraform-website/master/content/source/assets/images/logo-hashicorp.svg" alt="Terraform logo" title="Terraform Provider Jarvis" align="right" height="50" />
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply README.md:2]: Deleted.. That image doesn't work anymore

File: README.md [hunk 5]
```diff
@@ -1 +1,96 @@
...
+
+- Website: [terraform.io](https://terraform.io)
+- Tutorials: [learn.hashicorp.com](https://learn.hashicorp.com/terraform?track=getting-started#getting-started)
+- Forum: [discuss.hashicorp.com](https://discuss.hashicorp.com/c/terraform-providers/tf-indykite/)
+- Chat: [gitter](https://gitter.im/hashicorp-terraform/Lobby)
+- Mailing List: [Google Groups](http://groups.google.com/group/terraform-tool)
+
+The Terraform JARVIS provider is a plugin for Terraform that allows for the full
+lifecycle management of IndyKite JARVIS resources.
+This provider is maintained internally by the IndyKite JARVIS Provider team.
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply README.md]: No more JARVIS in files

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            exclude: ^indykite\/(utilities.go|resource_oauth2_client_test.go)$
+          - id: pretty-format-json
+          - id: no-commit-to-branch
+            args: [--branch, master]
+
+    - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt.git
+      rev: 0.0.10
+      hooks:
+          - id: yamlfmt
+
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply .pre-commit-config.yaml:42]: Cool, addded and formatted TF files

File: indykite/common_schema.go [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+		Description:      `The display name for the instance. Can be updated without creating a new resource.`,
+	}
+}
+
+func descriptionSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:         schema.TypeString,
+		Optional:     true,
+		ValidateFunc: validation.StringLenBetween(0, 256),
+		Description:  `A text description of the service account. Must be less than or equal to 256 UTF-8 bytes.`,
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply indykite/common_schema.go]: Yep, fixed to more generic description message

File: indykite/jarvis_suite_test.go [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+	"github.com/golang/mock/gomock"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+)
+
+// #nosec G101:
+const (
+	// customerID is test.CustomerWonkaGID
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply indykite/jarvis_suite_test.go]: Gone

File: indykite/resource_ingest_maaping_test.go [hunk 9]
```diff
@@ -0,0 +1,435 @@
+// Copyright (c) 2022 IndyKite
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply indykite/resource_ingest_maaping_test.go]: fixed

File: indykite/jarvis_suite_test.go [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+	appSpaceID = "gid:AAAAAmluZHlraURlgAABDwAAAAA"
+	// issuerID is test.IssuerWonkaFactoryGID
+	issuerID = "gid:AAAAD2luZHlraURlgAAEDwAAAAA"
+	// tenantID is test.TenantWonkaFactoryWonka1GID
+	tenantID = "gid:AAAAA2luZHlraURlgAADDwAAAAE"
+	// applicationID is test.ApplicationWonkaFactoryWonkaBarsGID
+	applicationID = "gid:AAAABGluZHlraURlgAACDwAAAAA"
+	// appAgentID is test.OPAAppAgent
+	appAgentID = "gid:AAAABWluZHlraURlgAAFDwAAAAA"
+	// appAgentCredID is test.WonkaOPAAppAgentCredID
```
[the remark this answers is not in the document]
Comment (arxeiss) [reply indykite/jarvis_suite_test.go]: Removed

File: example/main.tf (hunk 1 above)
Comment (arxeiss) [reply example/main.tf:183]: Keeping for now, might use in future sprints when testing. If not, will remove then

File: example/main.tf (hunk 2 above)
Comment (arxeiss) [reply example/main.tf:247]: same as above

File: indykite/provider.go (hunk 3 above)
Comment (arxeiss) [reply indykite/provider.go]: Remove

**Review by maaland — APPROVED** [review]

**Review by marvinkite — APPROVED** [review]

**Pull request merged**